scripts/src/main/resources/scripts/command/aws

```diff
@@ -12,85 +12,72 @@ fi
+source "$(dirname "${0}")"/../commandlet-cli
 
-# OS independent check if aws is installed.
-function findAws() {
-  AWS="aws"
-  AWS_INSTALLED="no"
-  if command -v aws &> /dev/null
+function setConfig() {
+  local aws_config_file="${DEVON_IDE_HOME}/conf/.aws"
+  local aws_config_export="export AWS_CONFIG_FILE=${aws_config_file}"
+  if ! grep -q "${aws_config_export}" "${DEVON_IDE_HOME}/conf/devon.properties"
   then
-    AWS_INSTALLED="yes"
-  elif doIsWindows
-  then
-    local aws_win
-    aws_win="${PROGRAMFILES}"
-    aws_win="$(cygpath "${aws_win}")"
-    aws_win="${aws_win}/Amazon/AWSCLIV2/aws"
-    if [ -f "${aws_win}" ]
-    then
-      AWS_INSTALLED="yes"
-    fi
-    AWS="${aws_win}"
+    doRunCommand "${aws_config_export}"
+    echo -e "\n${aws_config_export}" >> "${DEVON_IDE_HOME}/conf/devon.properties"
+    doEcho "Location of AWS's configuration file is set to ${aws_config_file}"
   fi
-  TOOL_VERSION_COMMAND="'${AWS}' --version"
 }
 
-findAws
 # shellcheck source=scripts/commandlet-cli
 source "$(dirname "${0}")"/../commandlet-cli
 
 # Call aws with specified arguments.
 function doRun() {
   doSetup silent
-  findAws
-  doRunCommand "'${AWS}' ${*}"
+  doRunCommand "${AWS_HOME}/aws ${*}"
 }
 
-# $1: path_to_package
-function doPackageInstall() {
-  local path_to_package="${1}"
+function doSetup() {
+  # workaround as we are downloading the installer and not installing the software itself
+  TOOL_VERSION_COMMAND="-"
+  doInstall "aws" "${AWS_VERSION}" "${1}" "" "${AWS_HOME}"
+  ERR="${?}"
+  local version_to_install
+  version_to_install=$(cat "${AWS_HOME}"/.devon.software.version)
   if doIsWindows
   then
-    doEcho "Installing AWS for Windows..."
-    windows_path_to_package=$(cygpath -w "${path_to_package}")
-    powershell.exe -Command "Start-Process msiexec.exe -verb runas -Wait -ArgumentList '/I ${windows_path_to_package}\aws-*-windows.msi /quiet'"
+    if [ "${ERR}" = 0 ]
+    then
+      mv "${AWS_HOME}"/Amazon/AWSCLIV2/* "${AWS_HOME}"
+      [ -d "${AWS_HOME}"//Amazon/ ] && rm -rf "${AWS_HOME}"/Amazon/
+      [ -f "${AWS_HOME}"/aws-"${version_to_install}"-windows.msi ] && rm -rf "${AWS_HOME}"/aws-"${version_to_install}"-windows.msi
+    fi
   elif doIsMacOs
   then
-    doEcho "Installing AWS for MacOS..."
-    sudo installer -pkg "${path_to_package}"/aws-"${version}"-mac.pkg -target /
+    if [ "${ERR}" = 0 ]
+    then
+      xar -C "${AWS_HOME}" -xf "${AWS_HOME}"/aws-"${version_to_install}"-mac.pkg
+      tar -C "${AWS_HOME}" -xzvf "${AWS_HOME}"/aws-cli.pkg/Payload
+      rm -rf "${AWS_HOME}"/Distribution "${AWS_HOME}"/Resources "${AWS_HOME}"/aws-cli.pkg
+      [ ! -h "${AWS_HOME}/aws" ] && ln -s "${AWS_HOME}"/aws-cli/aws "${AWS_HOME}"/aws
+      [ ! -h "${AWS_HOME}/aws_completer" ] && ln -s "${AWS_HOME}"/aws-cli/aws_completer "${AWS_HOME}"/aws_completer
+      [ -f "${AWS_HOME}"/aws-"${version_to_install}"-mac.pkg ] && rm "${AWS_HOME}"/aws-"${version_to_install}"-mac.pkg
+    fi
```

Comment:
I would love to see all such magic in do install but in a generic way (autodelete the pointless MSI copy that msiexec is creating for whatever reason that nobody can understand, also add extraction of pkg files to doExtract, etc.).
However, to keep it simple, I will create a new story for this.

Reply:
I created #1010 for this